src/config.ts

```diff
@@ -178,35 +178,117 @@ export class ConfigManager
             });
         }
     }
-    suggestedSettings(): any
+
+    autoApplyRubySettings() : boolean {
+        return this._vscode.getConfiguration().get("rock.applyRubySettings", true);
+    }
+
+    autoApplyCPPSettings() : boolean {
+        return this._vscode.getConfiguration().get("rock.applyCPPSettings", true);
+    }
+
+    autoApplySettings() {
+        if (this._workspaces.size !== 1) {
+            return;
+        }
+
+        if (this.autoApplyRubySettings()) {
+            this.applyRubySettings(vscode.ConfigurationTarget.Workspace);
+        }
+        if (this.autoApplyCPPSettings()) {
+            this.applyCPPSettings(vscode.ConfigurationTarget.Workspace);
+        }
+    }
+
+    suggestedEditorSettings(scope: vscode.ConfigurationTarget): any
     {
         return {
-            "C_Cpp.intelliSenseEngine": "Default",
-            "C_Cpp.intelliSenseEngineFallback": "Enabled",
             "editor.detectIndentation": false,
             "editor.insertSpaces": true,
-            "editor.rulers": [80],
+            "editor.rulers": [80, 90],
             "editor.tabSize": 4,
             "editor.trimAutoWhitespace": true,
             "files.trimTrailingWhitespace": true,
-            "git.countBadge": "tracked",
-            "ruby.lint": {
-                "rubocop": {
-                    "except": [
-                        "Layout/IndentationWidth",
```

Comment:
Isn't removing this gonna make rubocop complain about every single line across the entire Rock framework? I believe the default indention width is 2.

Reply:
Not if you https://github.com/rock-core/rock.vscode-package_set/pull/1 gets merged first.

Reply:
Pretty neat.. Nice 👍